app/services/submission_service.rb

```diff
@@ -15,6 +15,19 @@ def create_submission(submission_params, gravity_user_id)
       ]
       user = User.find_or_create_by!(gravity_user_id: gravity_user_id)
       create_params = submission_params.merge(user_id: user.id)
+
+      artist = Gravity.client.artist(id: submission_params[:artist_id])._get
+      unless artist[:target_supply]
+        create_params =
+          create_params.merge(
+            {
+              state: 'rejected',
+              rejection_reason: 'Not Target Supply',
+              rejected_at: Time.now.utc
+            }
+          )
+      end
```

Comment:
here -> update create params with state rejected if artist is non-target-supply

Reply:
I think it would be better to wrap this in a method with a self-explanatory name